File: src/main/java/play/pluv/playlist/application/dto/PlayListMusicResponse.java
package play.pluv.playlist.application.dto;

import java.util.List;
import lombok.Builder;
import play.pluv.playlist.domain.PlayListMusic;

@Builder
public record PlayListMusicResponse(
    String title, String artistNames, String isrcCode, String imageUrl
) {

  public static List<PlayListMusicResponse> createList(final List<PlayListMusic> musics) {
    return musics.stream()
        .map(PlayListMusicResponse::from)
        .toList();
  }

  private static PlayListMusicResponse from(final PlayListMusic playListMusic) {
    return PlayListMusicResponse.builder()
        .title(playListMusic.getTitle())
        .artistNames(String.join(",", playListMusic.getArtistNames()))
        .imageUrl(playListMusic.getImageUrl())
        .isrcCode(playListMusic.getIsrcCode().orElse(null))
        .build();
  }
}

File: src/main/kotlin/play/pluv/playlist/application/dto/PlayListMusicResponse.kt
package play.pluv.playlist.application.dto

import play.pluv.playlist.domain.PlayListMusic

data class PlayListMusicResponse(
    val title: String, val artistNames: String, val isrcCode: String?, val imageUrl: String
) {
    companion object {
        fun createList(musics: List<PlayListMusic>): List<PlayListMusicResponse> {
            return musics.map { from(it) }
        }

        private fun from(playListMusic: PlayListMusic): PlayListMusicResponse {
            return PlayListMusicResponse(
                title = playListMusic.title,
                artistNames = playListMusic.artistNames.joinToString(","),
                isrcCode = playListMusic.isrcCode,
                imageUrl = playListMusic.imageUrl
            )
        }
    }
}
